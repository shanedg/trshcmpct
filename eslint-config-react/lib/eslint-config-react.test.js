const eslint = require('eslint');

const ESLint = eslint.ESLint;

describe('eslint-config', () => {
  let eslintApi;

  beforeAll(() => {
    eslintApi = new ESLint({
      overrideConfigFile: 'lib/eslint-config-react.js',
      useEslintrc: false,
    });
  });

  describe('calculated config snapshots', () => {
    let config;

    beforeAll(async () => {
      // This file doesn't actually have to exist and the path doesn't matter
      // since we override the config file and useEslintrc is false.
      config = await eslintApi.calculateConfigForFile('SomeComponent.jsx');
    });

    it('parserOptions', () => {
      expect(config.parserOptions).toMatchSnapshot();
    });

    it('rules', () => {
      expect(config.rules).toMatchSnapshot();
    });
  });

});
