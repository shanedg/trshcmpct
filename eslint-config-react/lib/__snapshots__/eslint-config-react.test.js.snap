// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint-config calculated config snapshots parserOptions 1`] = `
Object {
  "ecmaFeatures": Object {
    "jsx": true,
  },
}
`;

exports[`eslint-config calculated config snapshots rules 1`] = `
Object {
  "jsx-a11y/alt-text": Array [
    "error",
  ],
  "jsx-a11y/anchor-has-content": Array [
    "error",
  ],
  "jsx-a11y/anchor-is-valid": Array [
    "error",
  ],
  "jsx-a11y/aria-activedescendant-has-tabindex": Array [
    "error",
  ],
  "jsx-a11y/aria-props": Array [
    "error",
  ],
  "jsx-a11y/aria-proptypes": Array [
    "error",
  ],
  "jsx-a11y/aria-role": Array [
    "error",
  ],
  "jsx-a11y/aria-unsupported-elements": Array [
    "error",
  ],
  "jsx-a11y/autocomplete-valid": Array [
    "error",
  ],
  "jsx-a11y/click-events-have-key-events": Array [
    "error",
  ],
  "jsx-a11y/control-has-associated-label": Array [
    "off",
    Object {
      "ignoreElements": Array [
        "audio",
        "canvas",
        "embed",
        "input",
        "textarea",
        "tr",
        "video",
      ],
      "ignoreRoles": Array [
        "grid",
        "listbox",
        "menu",
        "menubar",
        "radiogroup",
        "row",
        "tablist",
        "toolbar",
        "tree",
        "treegrid",
      ],
      "includeRoles": Array [
        "alert",
        "dialog",
      ],
    },
  ],
  "jsx-a11y/heading-has-content": Array [
    "error",
  ],
  "jsx-a11y/html-has-lang": Array [
    "error",
  ],
  "jsx-a11y/iframe-has-title": Array [
    "error",
  ],
  "jsx-a11y/img-redundant-alt": Array [
    "error",
  ],
  "jsx-a11y/interactive-supports-focus": Array [
    "error",
    Object {
      "tabbable": Array [
        "button",
        "checkbox",
        "link",
        "searchbox",
        "spinbutton",
        "switch",
        "textbox",
      ],
    },
  ],
  "jsx-a11y/label-has-associated-control": Array [
    "error",
  ],
  "jsx-a11y/label-has-for": Array [
    "off",
  ],
  "jsx-a11y/media-has-caption": Array [
    "error",
  ],
  "jsx-a11y/mouse-events-have-key-events": Array [
    "error",
  ],
  "jsx-a11y/no-access-key": Array [
    "error",
  ],
  "jsx-a11y/no-autofocus": Array [
    "error",
  ],
  "jsx-a11y/no-distracting-elements": Array [
    "error",
  ],
  "jsx-a11y/no-interactive-element-to-noninteractive-role": Array [
    "error",
    Object {
      "canvas": Array [
        "img",
      ],
      "tr": Array [
        "none",
        "presentation",
      ],
    },
  ],
  "jsx-a11y/no-noninteractive-element-interactions": Array [
    "error",
    Object {
      "alert": Array [
        "onKeyUp",
        "onKeyDown",
        "onKeyPress",
      ],
      "body": Array [
        "onError",
        "onLoad",
      ],
      "dialog": Array [
        "onKeyUp",
        "onKeyDown",
        "onKeyPress",
      ],
      "handlers": Array [
        "onClick",
        "onError",
        "onLoad",
        "onMouseDown",
        "onMouseUp",
        "onKeyPress",
        "onKeyDown",
        "onKeyUp",
      ],
      "iframe": Array [
        "onError",
        "onLoad",
      ],
      "img": Array [
        "onError",
        "onLoad",
      ],
    },
  ],
  "jsx-a11y/no-noninteractive-element-to-interactive-role": Array [
    "error",
    Object {
      "fieldset": Array [
        "radiogroup",
        "presentation",
      ],
      "li": Array [
        "menuitem",
        "option",
        "row",
        "tab",
        "treeitem",
      ],
      "ol": Array [
        "listbox",
        "menu",
        "menubar",
        "radiogroup",
        "tablist",
        "tree",
        "treegrid",
      ],
      "table": Array [
        "grid",
      ],
      "td": Array [
        "gridcell",
      ],
      "ul": Array [
        "listbox",
        "menu",
        "menubar",
        "radiogroup",
        "tablist",
        "tree",
        "treegrid",
      ],
    },
  ],
  "jsx-a11y/no-noninteractive-tabindex": Array [
    "error",
    Object {
      "allowExpressionValues": true,
      "roles": Array [
        "tabpanel",
      ],
      "tags": Array [],
    },
  ],
  "jsx-a11y/no-redundant-roles": Array [
    "error",
  ],
  "jsx-a11y/no-static-element-interactions": Array [
    "error",
    Object {
      "allowExpressionValues": true,
      "handlers": Array [
        "onClick",
        "onMouseDown",
        "onMouseUp",
        "onKeyPress",
        "onKeyDown",
        "onKeyUp",
      ],
    },
  ],
  "jsx-a11y/role-has-required-aria-props": Array [
    "error",
  ],
  "jsx-a11y/role-supports-aria-props": Array [
    "error",
  ],
  "jsx-a11y/scope": Array [
    "error",
  ],
  "jsx-a11y/tabindex-no-positive": Array [
    "error",
  ],
  "react-hooks/exhaustive-deps": Array [
    "warn",
  ],
  "react-hooks/rules-of-hooks": Array [
    "error",
  ],
  "react/display-name": Array [
    2,
  ],
  "react/jsx-key": Array [
    2,
  ],
  "react/jsx-no-comment-textnodes": Array [
    2,
  ],
  "react/jsx-no-duplicate-props": Array [
    2,
  ],
  "react/jsx-no-target-blank": Array [
    2,
  ],
  "react/jsx-no-undef": Array [
    2,
  ],
  "react/jsx-uses-react": Array [
    2,
  ],
  "react/jsx-uses-vars": Array [
    2,
  ],
  "react/no-children-prop": Array [
    2,
  ],
  "react/no-danger-with-children": Array [
    2,
  ],
  "react/no-deprecated": Array [
    2,
  ],
  "react/no-direct-mutation-state": Array [
    2,
  ],
  "react/no-find-dom-node": Array [
    2,
  ],
  "react/no-is-mounted": Array [
    2,
  ],
  "react/no-render-return-value": Array [
    2,
  ],
  "react/no-string-refs": Array [
    2,
  ],
  "react/no-unescaped-entities": Array [
    2,
  ],
  "react/no-unknown-property": Array [
    2,
  ],
  "react/no-unsafe": Array [
    0,
  ],
  "react/prop-types": Array [
    2,
  ],
  "react/react-in-jsx-scope": Array [
    2,
  ],
  "react/require-render-return": Array [
    2,
  ],
}
`;
